src/helpers.ts

```diff
@@ -55,6 +55,37 @@ export class TurbopufferError extends Error {
   }
 }
 
+export function getUpdatedUrlPath(
+  baseUrl: string,
+  path: string,
+  query?: Record<string, string | undefined>,
+) {
+  // https://developer.mozilla.org/en-US/docs/Web/API/URL_API/Resolving_relative_references
+  // if baseUrl doesn't end in /, add one to make it behave
+  // like a directory so the next path part is appended.
+  // if there are multiple / appended, ensure all but one get removed.
+  let updatedBaseUrl = baseUrl.replace(/\/+$/, "");
+  updatedBaseUrl += "/";
```

Comment:
could do this and the previous operation in one call `baseUrl.replace(/\/*$/, "/")`

Reply:
this only works if the baseUrl already ends in a `/` (i'll add a comment to make it clear!)